File: sand/labs/lab2/lab2.m


%% 2.0 Load data...

% either Axon Diameter vs Conduction Velocity
data = readtable('Hursh1936.csv');

% or log-Body Weight vs log-Brain Weight
data = readtable('BodyBrain.csv');

data = table2array(data);
x = data(:,1);
y = data(:,2);
n = size(data,1);

%% 2.1 Bootstrap to estimate the mean and sem of the x-variable

nboot = 10000;
bootstat = bootstrp(nboot,'mean',???);
histogram(???)

%% 2.2 Plot the data


%% 2.3 Linear regression

% linear regression
b = polyfit(???);

%% 2.4 Plot the data and fit


%% 2.5 Bootstrap to find error-bars for the model parameters

nboot = 1000;
for i=1:nboot
    sample_indices = randsample(n,n,true);
    bboot(i,:) = polyfit(???);
end

File: sand/labs/lab2/lab2.R


### 2.0 Load data...

# either Axon Diameter vs Conduction Velocity
data <- read.csv(file='Hursh1936.csv',header=FALSE);

# or log-Body Weight vs log-Brain Weight
#data <- read.csv('BodyBrain.csv',header=TRUE);


### 2.1 Bootstrap to estimate the mean and sem of the x-variable

library(boot)

nboot = 10000
mfun <- function(x,w) ???
bootobject <- boot(???, mfun, R=nboot, stype='w')

sd(???)
hist(???)

### 2.2 Plot the data

plot(???)

### 2.3 Linear regression

# linear regression
fit <- lm(???)


## 2.4 Plot the data and fit

plot(???)
lines(???)


## 2.5 Bootstrap to find error-bars for the model parameters

nboot = 100
ffun <- function(d,i) {
  f<-lm(???)
  f$coefficients
}
bootfit <- boot(data, ffun, R=nboot, stype='i')
for (i in 1:10) {
  lines(???)
}